### PR TITLE
MAT-8150 and 8151: Value Set Search - Definition Version

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-8150](https://jira.cms.gov/browse/MAT-8150) and [MAT-8151](https://jira.cms.gov/browse/MAT-8151)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/CqlBuilderPanel/ValueSets/Search/Search.tsx [hunk 1]
```diff
@@ -149,6 +163,16 @@ export default function Search(props: SearchProps) {
...
+                      placeHolder={
+                        value == "version" && formik.values.url.length === 0
+                          ? "OID/URL must be entered first"
+                          : ""
+                      }
+                      disabled={
+                        value == "version"
+                          ? formik.values.url.length === 0
+                          : false
+                      }
```
Comment (mcmcphillips) [inline src/CqlBuilderPanel/ValueSets/Search/Search.tsx]: In general you do not want to use == in javascript as it could cause unforseen issues since it's not strict equality. 

this check
formik.values.url.length === 0
could be the same as !formik.values.url to account for edge case when value is undefined and not a string. 
Doesn't seem to hit this, but it's a safe way to do it other places

File: src/CqlBuilderPanel/ValueSets/Search/Search.tsx [hunk 2]
```diff
@@ -160,6 +184,14 @@ export default function Search(props: SearchProps) {
...
                   label={SEARCH_MAP[value]}
                   trimField={trimField}
+                  placeHolder={
+                    value == "version" && formik.values.url.length === 0
+                      ? "OID/URL must be entered first"
+                      : ""
+                  }
+                  disabled={
+                    value == "version" ? formik.values.url.length === 0 : false
+                  }
```
Comment (mcmcphillips) [inline src/CqlBuilderPanel/ValueSets/Search/Search.tsx]: avoid ==, only ===. 
Same as comment above, you can check to see if a value is truthy like !formik.values.url

**Review by mcmcphillips — DISMISSED** [review]

**Review by mcmcphillips — APPROVED** [review]
